www/js/customEventHandler.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * since react doesn't quite support custom events, writing our own handler
+ * having the ability to broadcast and emit events prevents files from being tightly coupled
+ * if we want something else to happen when an event is emitted, we can just listen for it
+ * instead of having to change the code at the point the event is emitted
+ *
+ * looser coupling = point of broadcast doesn't 'know' what is triggered by that event
+ * leads to more extensible code
+ * consistent event names help us know what happens when
+ *
+ * code based on: https://blog.logrocket.com/using-custom-events-react/
+ */
+
+import { logDebug } from './plugin/logger';
+
+/**
+ * central source for event names
+ */
+export const EVENT_NAMES = {
+  CLOUD_NOTIFICATION_EVENT: 'cloud:push:notification',
+  CONSENTED_EVENT: 'data_collection_consented',
+  INTRO_DONE_EVENT: 'intro_done',
+};
```

Comment:
nitpicky, but might it be nice to rename and make these more concise?

Wherever these are used, we have to write out `EVENT_NAMES.CLOUD_NOTIFICATION_EVENT`. It seems redundant to repeat the word 'event' and I think it's self-evident that these are the names of the events
Could we rename to `EVENTS.CLOUD_NOTIFICATION`?